cmd/deploy.go

```diff
@@ -386,33 +379,58 @@ func installHelmRelease(ctx context.Context, helmClient *helm.Client, releaseNam
 	return err
 }
 
-func validateInstall(ctx context.Context, kubeClient *k8s.Client, namespace, appVersion string, auth0Token *auth.Auth0Token, clusterName string, deployableNodesCount int, persistentStorage bool, sentryHelmContext *sentry_utils.HelmContext) error {
+func validateInstall(ctx context.Context, kubeClient *k8s.Client, namespace, appVersion string, clusterName string, deployableNodesCount int, persistentStorage bool, sentryHelmContext *sentry_utils.HelmContext) (bool, error) {
 	var err error
 
 	ui.GlobalWriter.PrintlnWithPrefixln("Validating groundcover installation:")
 
 	if persistentStorage {
 		if err = waitForPvcs(ctx, kubeClient, namespace, sentryHelmContext); err != nil {
-			return err
+			return false, err
 		}
 	}
 
 	if err = waitForPortal(ctx, kubeClient, namespace, appVersion, sentryHelmContext); err != nil {
-		return err
+		return false, err
 	}
 
 	if err = waitForAlligators(ctx, kubeClient, namespace, appVersion, deployableNodesCount, sentryHelmContext); err != nil {
-		return err
+		return false, err
 	}
 
-	if !viper.IsSet(TOKEN_FLAG) {
-		apiClient := api.NewClient(auth0Token)
-		if err = apiClient.PollIsClusterExist(ctx, clusterName); err != nil {
-			return err
-		}
+	return true, nil
+}
+
+func validateClusterRegistered(ctx context.Context, clusterName string) (bool, error) {
+	var err error
+
+	var auth0Token auth.Auth0Token
+	if err = auth0Token.Load(); err != nil {
```

Comment:
I don't like allocating empty objects on the stack.
I think that `Load()` should return a new `*auth.Auth0Token`